packages/target-tauri/src-tauri/src/menus/tray_menu.rs

```diff
@@ -0,0 +1,79 @@
+use anyhow::Context;
+use strum::{AsRefStr, EnumString};
+use tauri::{
+    menu::{CheckMenuItem, Menu, MenuItem},
+    AppHandle, Manager, Wry,
+};
+use tauri_plugin_store::StoreExt;
+
+use crate::{
+    get_setting_bool_or,
+    settings::{NOTIFICATIONS, NOTIFICATIONS_DEFAULT},
+    state::main_window_channels::MainWindowEvents,
+    MainWindowChannels, TranslationState, TrayManager, CONFIG_FILE,
+};
+
+use super::menu_action::{impl_menu_conversion, MenuAction};
+
+#[derive(Debug, AsRefStr, EnumString)]
+pub(crate) enum TrayMenuAction {
+    Quit,
+    Show,
+    MuteNotifications,
+}
+impl_menu_conversion!(TrayMenuAction);
+
+impl MenuAction<'static> for TrayMenuAction {
+    async fn execute(self, app: &AppHandle) -> anyhow::Result<()> {
+        let main_window = app.get_window("main").context("main window not found")?;
+
+        match self {
+            TrayMenuAction::Quit => {
+                app.exit(0);
+            }
+            TrayMenuAction::Show => main_window.show()?,
```

Comment:
```suggestion
            TrayMenuAction::Show => {
                main_window.show()?;
                main_window.set_focus()?;
            },
```

I think we need to do the same for all the other `.show()` occurrences. Because if the window is just minimized and not hidden, then nothing will happen.